File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class TicketVolumePoint(BaseModel):
    date: str
    count: int


class TopIssueCategory(BaseModel):
    category: str
    count: int


class CountResponse(BaseModel):
    count: int


class AnalyticsSummaryResponse(BaseModel):
    ticket_volume_7d: list[TicketVolumePoint]
    avg_resolution_time_hours: float | None = None
    csat_average: float | None = None
    deflection_rate: float | None = None

File: frontend/src/types/analytics.ts
export interface TicketVolumePoint {
  date: string;
  count: number;
}

export interface TopIssueCategory {
  category: string;
  count: number;
}

export interface AnalyticsSummary {
  ticket_volume_7d: TicketVolumePoint[];
  avg_resolution_time_hours: number | null;
  csat_average: number | null;
  deflection_rate: number | null;
}
